specification/authorization/resource-manager/Microsoft.Authorization/2017-10-01-preview/authorization.json

```diff
@@ -685,7 +685,7 @@
           "type": "string",
           "description": "The role definition description."
         },
-        "roletype": {
+        "type": {
           "type": "string",
```

Comment:
From what we t6alked offline, this should be renamed to `roleName`

Reply:
since there are two fields with name "type" in the response that is sent by the service.
we cant name this as rolename,because it wouldnt match the property "type" and would miss out having the value like custom role/builtin role

Reply:
Okok, there was some confusion on my side.
The property name should be the very same that is returned by the service. In this case, if you want to still get the benefits from x-ms-client-flatten, there is another extension that can be used to override the property name in the SDK code without changing serialization. Look into [x-ms-client-name](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-client-name)